src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.Linux.cs

```diff
@@ -502,6 +511,9 @@ private void CancellationCallback()
 
                     _wdToPathMap.Clear();
                 }
+
+                // Unlock the root directory
+                Debug.Assert(Interop.Sys.FLock(_inotifyHandle, Interop.Sys.LockOperations.LOCK_UN) == 0);
```

Comment:
Also, I'm curious... why is the inotify handle the right one to lock?  Is that actually just a file descriptor for the underlying directory somehow?  And do we need to be explicitly unlocking it, or does our existing disposal of the _inotifyHandle suffice?